build.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use regex::Regex;
+use std::fs::File;
+use std::fs::{read_dir, remove_file};
+use std::io::{Read, Write};
+use std::process::Command;
+
+fn main() {
+    check_protoc_version();
+    let _ = remove_file("src/lib.rs");
+
+    let file_names: Vec<_> = read_dir("proto")
+        .expect("Couldn't read proto directory")
+        .map(|e| {
+            format!(
+                "proto/{}",
+                e.expect("Couldn't list file").file_name().to_string_lossy()
+            )
+        })
+        .collect();
+    let file_names: Vec<_> = file_names.iter().map(|s| &**s).collect();
+
+    for f in &file_names {
+        println!("cargo:rerun-if-changed={}", f);
+    }
+
+    generate_rust_files(file_names);
+
+    let mod_names: Vec<_> = read_dir("src")
+        .expect("Couldn't read src directory")
+        .filter_map(|e| {
+            let file_name = e.expect("Couldn't list file").file_name();
+            file_name
+                .to_string_lossy()
+                .split(".rs")
+                .next()
+                .map(|n| n.to_owned())
+        })
+        .collect();
+    
+    replace_read_unknown_fields(&mod_names);
```

Comment:
How about sort it first?

Reply:
Do you mean sort `mod_names` so that the `mod` declarations in lib.rs are in alphabetical order? Or for some other purpose?

Reply:
Yes, exactly.

Reply:
@BusyJay I think this PR is kind of obsolete due to  #349.

Reply:
I was planning on landing this first, then #349 as a follow-up

Reply:
Cool!